_posts/2022-02-25-your-personal-glean-data-pipeline.md

```diff
@@ -0,0 +1,63 @@
+---
+permalink: "/{{ year }}/{{ month }}/{{ day }}/your-personal-glean-data-pipeline"
+title: "This Week in Glean: Your personal Glean data pipeline"
+published_date: "2022-02-25 12:01:00 +0100"
+layout: post.liquid
+data:
+  route: blog
+  tags:
+    - mozilla
+excerpt: |
+  On February 11th, 2022 I presented a lightning talk titled
+  "Your personal Glean data pipeline", presenting a little side project for
+  ingesting, transforming and analyzing data collected from Glean-powered applications myself.
+---
+
+(“This Week in Glean” is a series of blog posts that the Glean Team at Mozilla is using to try to communicate better about our work.
+They could be release notes, documentation, hopes, dreams, or whatever: so long as it is inspired by Glean.)
+All "This Week in Glean" blog posts are listed in the [TWiG index](https://mozilla.github.io/glean/book/appendix/twig.html)
+(and on the [Mozilla Data blog](https://blog.mozilla.org/data/category/glean/)).
+This article is [cross-posted on the Mozilla Data blog][datablog].
+
+[datablog]: https://blog.mozilla.org/data/TODO
+
+---
+
+On February 11th, 2022 we hosted a Data Club Lightning Talk session.
+There I presented my small side project of setting up a minimal data pipeline & data storage parts for Glean.
+
+The premise:
+
+> Can I build and run a small pipeline & data server to collect telemetry data from my own usage of tools?
+
+To which I can now answer: Yes, it's possible.
+The complete ingestion server is a couple hundred lines of Rust code.
+It's able to receive pings conforming to the Glean ping schema, transform them and store it into an SQLite database.
+It's very robust, not crashing once on me (except when I created an infinite loop within it).
+
+You can watch the lightning talk here:
+
+<br>
+
+<iframe width="800" height="450" src="https://www.youtube.com/embed/V5FgVbxm-cc" title="YouTube video player" frameborder="0" allow="picture-in-picture" allowfullscreen></iframe>
+
+[report]: https://tmp.fnordig.de/priv/mozilla/data-pipeline/report.html
+
+Instead of creating some slides for the talk I created an interactive report.
+The full report [can be read online](https://data.fnordig.de/glean/report.html).
```

Comment:
In the report the "Guess when the most recent TWIG was published" graph no longer includes the time when the most-recent TWiG was published : ) (( You could expand the default value in the number field to 28 to account for the march of time ))